api/types/resource.go

```diff
@@ -67,6 +67,16 @@ type ResourceWithSecrets interface {
 	WithoutSecrets() Resource
 }
 
+// ResourceWithOrigin provides information on the origin of the resource
+// (config-file, defaults, ...).
+type ResourceWithOrigin interface {
+	Resource
+	// IsFromConfigFile returns true if the resource originates from config file.
+	IsFromConfigFile() bool
+	// IsFromConfigFile returns true if the resource originates from defaults.
+	IsFromDefaults() bool
```

Comment:
It was my intention to *not* make the interface universal in this sense to make sure our code always handles only those origin values that have an explicitly defined meaning. Since `Orgin()` would return just a plain string it could happen to be compared to an unrelated string value, either in an ad hoc manner (which we should discourage) or by accident (which we should prevent as well).

Reply:
How about returning a typed string?
`type Origin string` and make the constants in `api/types/constants.go` use that.
That way you'd need to explicitly convert an unrelated string for comparison.